js/main.js

```diff
@@ -496,32 +527,63 @@ function renderAllLiked() {
 }
 
 function handleImageClick(event) {
-  if (event.target.tagName === 'IMG' && event.target.id !== 'detail-image') {
-    // Image was clicked, now see it in detail
+  if (event.target.tagName === 'IMG' && !['detail-image', 'delete-image'].includes(event.target.id)) {
+    // Check if Delete Mode is active
+    // In theory, Delete Mode cannot be active when the bottom sheet is closed
+    // We don't need to worry about clicking display-image when Delete Mode is on
+    if (deleteMode === false) {
+      // Delete Mode off, show detail modal
+
+      // Put the selected image's data into data.viewingInDetail
+      if (event.target.id === 'display-image') {
+        // clicked on the image being decided
+        data.viewingInDetail = displayArtObj;
+      } else {
+        // clicked on an image in the Likes gallery
+        for (var i = 0; i < data.likedObjects.length; i++) {
+          if (String(data.likedObjects[i].objectID) === String(event.target.getAttribute('objectId'))) {
+            data.viewingInDetail = data.likedObjects[i];
+          }
+        }
+      }
 
-    // find image object of image that was clicked and put it in viewingInDetail
-    if (event.target.id === 'display-image') {
-      data.viewingInDetail = displayArtObj;
-    } else {
-      for (var i = 0; i < data.likedObjects.length; i++) {
-        if (String(data.likedObjects[i].objectID) === String(event.target.getAttribute('objectId'))) {
-          data.viewingInDetail = data.likedObjects[i];
+      // Point detail img to the selected image's url
+      // Setting requestFullSize to false, images are VERY large
+      addImageToImg(data.viewingInDetail, $detailModalImage, false);
+      $detailModalContainer.classList.remove('hidden');
+    } else if (deleteMode === true) {
+      // Delete Mode on, show delete modal
+
+      // Put the selected image's data into data.deleting
+      for (var d = 0; d < data.likedObjects.length; d++) {
+        if (String(data.likedObjects[d].objectID) === String(event.target.getAttribute('objectId'))) {
+          data.deleting = data.likedObjects[d];
```

Comment:
When you find the matching id, include a `break` to stop further iteration through the array.

```js
        if (String(data.likedObjects[d].objectID) === String(event.target.getAttribute('objectId'))) {
          data.deleting = data.likedObjects[d];
          break;
```

Reply:
I've added `break` to both the Liked search and the To Delete search, and I'll keep that in mind for similar loops in the future.